### PR TITLE
color-filter

### PR DESCRIPTION
**Pull request by lindsay-jin**

### CODE REVIEW

File: client/src/pages/Catalog.tsx [hunk 1]
```diff
@@ -1,24 +1,66 @@
...
   const [isSearching, setIsSearching] = useState(false);
   const [error, setError] = useState<unknown>();
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectColors, setSelectColors] = useState<string[]>([]);
   const { categoryId, subcategoryId } = useParams<{
     categoryId: string;
     subcategoryId: string;
   }>();
 
+  const filteredProducts = products?.filter((product) => {
```
Comment (RobertGardner) [inline client/src/pages/Catalog.tsx:57]: Very well done!

The solution not only works very well, but the code is beautiful.

File: client/src/pages/Catalog.tsx [hunk 2]
```diff
@@ -55,10 +97,80 @@ export function Catalog() {
...
+
+  function handleColorClick(color) {
+    setSelectColors((prevColors) => {
+      const isColorSelected = prevColors.includes(color);
+      if (isColorSelected) {
+        return prevColors.filter((c) => c !== color);
+      } else {
+        return [...prevColors, color];
+      }
+    });
```
Comment (RobertGardner) [inline client/src/pages/Catalog.tsx:149]: Once again, beautifully done.

File: client/src/pages/Catalog.tsx [hunk 3]
```diff
@@ -55,10 +97,80 @@ export function Catalog() {
...
+  }
+
   return (
     <>
+      <div className="flex justify-end mb-3">
```
Comment (RobertGardner) [inline client/src/pages/Catalog.tsx:201]: This is getting pretty long. At some point you might want to consider breaking it into smaller components. That's optional, of course, and can be done after you have everything else working.

File: server/server.ts [hunk 4]
```diff
@@ -72,7 +72,10 @@ app.get(
     `;
       const params = [categoryId, subcategoryId];
       const result = await db.query(sql, params);
-      const products = result.rows;
+      const products = result.rows.map((product) => ({
+        ...product,
+        colors: product.colors ? JSON.parse(product.colors) : [],
```
Comment (RobertGardner) [inline server/server.ts:77]: Nice.

**Review by RobertGardner — APPROVED** [review]
You really have become one of the best developers in the cohort. I love looking over your code!

File: client/src/pages/Catalog.tsx (hunk 1 above)
Comment (lindsay-jin) [reply client/src/pages/Catalog.tsx:57]: Thank you! I appreciate all your comments!

**Pull request merged**